pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -1360,14 +1388,83 @@ impl RemoteTimelineClient {
     }
 }
 
+pub fn remote_timelines_path(tenant_id: &TenantId) -> RemotePath {
+    let path = format!("tenants/{tenant_id}/{TIMELINES_SEGMENT_NAME}");
+    RemotePath::from_string(&path).expect("Failed to construct path")
```

Comment:
I'm generally leaning away from PathBuf for remote path stuff -- it isn't a good fit for for S3 keys, because they're just  unicode strings, and while .join() is handy for "/" separate things, it's awkward when assembling other parts of a key (like the generation suffix), and awkward when doing prefix-listing (because prefixes don't have to break at a separator).

At some point we should make `RemotePath` wrap String instead of PathBuf